### PR TITLE
Add a blog for prod-mode vs dev-mode

### PR DESCRIPTION
**Pull request by masayag**
The purpose of this PR is to provide information on the differences between prod-mode and dev-mode.
Related to https://github.com/janus-idp/backstage-plugins/issues/1815 

Signed-off-by: Moti Asayag <masayag@redhat.com>

rh-pre-commit.version: 2.3.0
rh-pre-commit.check-secrets: ENABLED

### CODE REVIEW

**Review by batzionb — APPROVED** [review]

File: content/blog/devmode-vs-prodmode.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+* **Persistence of Running Workflow Instances**: Ensuring the persistence of workflow instances is critical. In production, all workflow data is stored and maintained even if the orchestrator restarts, preventing data loss.
+* **Event Handling Reliability**: Reliable handling of events is essential for maintaining workflow integrity and ensuring that all triggers and actions occur as expected.
+* **Scalability**: The system must be capable of scaling up to handle increasing workloads, allowing for the addition of resources as demand grows.
+* **Updates Through Pipelines**: Workflows can be updated and managed through continuous integration/continuous deployment (CI/CD) pipelines, facilitating smooth and efficient updates.
+* **Externalizing Credentials**: Credentials are managed outside the workflow configuration to enhance security and simplify updates.
+* **Runtime Isolation**: Each workflow runs in its isolated environment, preventing any interference between workflows.
+* **Authorization and Administration**: Robust mechanisms are in place for authorizing and administering workflow deployments, ensuring that only authorized personnel can make changes.
```
Comment (batzionb) [inline content/blog/devmode-vs-prodmode.md:18]: how does the authorization work? you mean the same as any k8s CR?

File: content/blog/devmode-vs-prodmode.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+date: 2024-06-30
+title: Production Mode vs. Dev Mode
+---
+
+When setting up workflow orchestration, it's crucial to understand the differences between production mode and development (dev) mode, particularly in terms of infrastructure requirements. This distinction ensures that workflows are efficiently managed and executed based on their intended use case. Here, we'll explore these differences, focusing on the infrastructure required to run workflows in each mode.
+
+# Production Mode
+Production mode is tailored for environments where stability, reliability, and scalability are paramount. The Orchestrator Helm chart or Orchestrator operator is designed specifically to meet the demanding requirements of production environments. Key requirements include:
```
Comment (gciavarrini) [inline content/blog/devmode-vs-prodmode.md:9]: Links to both?

File: content/blog/devmode-vs-prodmode.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+# Development (Dev) Mode
+Dev mode, as the name implies, is optimized for development purposes. It allows developers to experiment with the Orchestrator plugins without the need for a full deployment process or access to a Kubernetes (K8s) or OpenShift (OCP) cluster. Characteristics of dev mode include:
+
+* **Ephemeral Workflows**: Workflows run in an ephemeral mode, meaning that they are temporary and do not persist after the container restarts. This is suitable for development and testing.
+* **No Persistence**: In dev mode, there is no persistence for running workflow instances. All instance information is lost after a container restart, making it ideal for short-running or non-critical workflows.
+* **Development Focus**: Dev mode is designed for developers to gain experience and test workflows without the overhead of a full production environment.
+* **Hot-deployment of Workflows**: Developers can deploy new workflows by simply placing the workflow files in a designated folder, enabling rapid iteration and testing.
```
Comment (gciavarrini) [inline content/blog/devmode-vs-prodmode.md:26]: Link to a guide that explains this?

**Review by gciavarrini — COMMENTED** [review]
What about adding a section/link that explains how to enable one mode or another?

But overall I like a lot this PR!

File: content/blog/devmode-vs-prodmode.md (hunk 1 above)
Comment (masayag) [reply content/blog/devmode-vs-prodmode.md:18]: yes, this is related to the workflow deployment (k8s RBAC to manage the sonataflow CRs)

File: content/blog/devmode-vs-prodmode.md (hunk 3 above)
Comment (masayag) [reply content/blog/devmode-vs-prodmode.md:26]: I'm not aware of such link. I'd be happy if someone with experience with dev-mode can create a blog to explain how to work with it.

File: content/blog/devmode-vs-prodmode.md (hunk 2 above)
Comment (masayag) [reply content/blog/devmode-vs-prodmode.md:9]: I can only include a link to the production mode which is the only installation method described in this website.
I'd be happy if someone with experience with dev-mode can create a blog to explain how to work with it and then add a reference to it from here.

File: content/blog/devmode-vs-prodmode.md (hunk 3 above)
Comment (gciavarrini) [reply content/blog/devmode-vs-prodmode.md:26]: Agree that the dev mode isn't explained. 
Created a Jira story to track this [FLPATH-1451](https://issues.redhat.com/browse/FLPATH-1451)

File: content/blog/devmode-vs-prodmode.md (hunk 2 above)
Comment (gciavarrini) [reply content/blog/devmode-vs-prodmode.md:9]: Agree that the dev mode isn't explained.
Created a Jira story to track this [FLPATH-1451](https://issues.redhat.com/browse/FLPATH-1451)

**Pull request merged**